apps/dialtone-documentation/docs/.vuepress/baseComponents/BaseIcon.vue

```diff
@@ -7,132 +7,149 @@
-import SpotMaleLaptopTyping from '@dialpad/dialtone/lib/dist/vue/spot/SpotMaleLaptopTyping.vue';
-import SpotFileUpload from '@dialpad/dialtone/lib/dist/vue/spot/SpotFileUpload.vue';
-import SpotBrowserTableGraph from '@dialpad/dialtone/lib/dist/vue/spot/SpotBrowserTableGraph.vue';
-import SpotBrowserListCallout from '@dialpad/dialtone/lib/dist/vue/spot/SpotBrowserListCallout.vue';
-import SpotMind from '@dialpad/dialtone/lib/dist/vue/spot/SpotMind.vue';
-import SpotBlankSpace from '@dialpad/dialtone/lib/dist/vue/spot/SpotBlankSpace.vue';
-import SpotVectorVortex from '@dialpad/dialtone/lib/dist/vue/spot/SpotVectorVortex.vue';
+<script setup>
+import { computed, ref } from 'vue';
+import { ICON_KINDS } from './constants.js';
+import CopyButton from './CopyButton.vue';
 
-export const ICON_KINDS = ['spot'];
-
-export default {
-  name: 'BaseIcon',
-  components: {
-    SpotPublish,
-    SpotWirelessScreenshare,
-    SpotFemaleLaptopTyping,
-    SpotMaleLaptopTyping,
-    SpotFileUpload,
-    SpotBrowserTableGraph,
-    SpotBrowserListCallout,
-    SpotMind,
-    SpotBlankSpace,
-    SpotVectorVortex,
+defineProps({
+  name: {
+    type: String,
+    required: true,
   },
 
-  props: {
-    name: {
-      type: String,
-      required: true,
-    },
-
-    file: {
-      type: String,
-      required: true,
-    },
-
-    desc: {
-      type: String,
-      default: '',
-    },
+  file: {
+    type: String,
+    required: true,
+  },
 
-    code: {
-      type: String,
-      default: '',
-    },
+  figmaLink: {
+    type: String,
+    required: true,
+  },
 
-    hidden: {
-      type: Boolean,
-      default: false,
-    },
+  desc: {
+    type: String,
+    default: '',
+  },
 
-    vue: {
-      type: String,
-      required: true,
-    },
+  code: {
+    type: String,
+    default: '',
+  },
 
-    kind: {
-      type: String,
-      required: true,
-      validator: (kind) => {
-        return ICON_KINDS.includes(kind);
-      },
-    },
+  hidden: {
+    type: Boolean,
+    default: false,
+  },
 
-    selected: {
-      type: Boolean,
-      default: false,
-    },
+  vue: {
+    type: String,
+    required: true,
   },
 
-  computed: {
-    isSpotKind () {
-      return this.kind === 'spot';
+  kind: {
+    type: String,
+    required: true,
+    validator: (kind) => {
+      return ICON_KINDS.includes(kind);
     },
+  },
 
-    selectedStatus () {
-      return this.selected ? 'yes' : 'no';
-    },
+  rawSvg: {
+    type: String,
+    required: true,
+  },
+});
 
-    cardFooterClass () {
-      return 'dialtone-icon-card__footer-spot-illustration';
-    },
+const selected = ref(false);
+const iconComponent = ref(null);
 
-    cardIconClass () {
-      return 'dialtone-icon-card__icon--autosize';
-    },
-  },
+const selectedStatus = computed(() => selected.value ? 'yes' : 'no');
+const cardIconClass = computed(() => 'dialtone-icon-card__icon--autosize');
```

Comment:
I know you just copied what was already there, but this seems pointless. We can just apply `dialtone-icon-card__icon--autosize` class directly to the element.